pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -2204,8 +2204,7 @@ impl Timeline {
                     .as_mut()
                     .unwrap()
                     .put_value(key, lsn, value, ctx)
-                    .await
-                    .map_err(CompactionError::Other)?;
+                    .await?;
```

Comment:
I believe this is not the only place that we convert the write errors but let's fix this for now and have future patches to correct all places.